### ISSUE TITLE
Zod method which alters the schema does not affect the OAS schema when it's already named

### ISSUE THREAD

**Issue by evertdespiegeleer** [asteasolutions/zod-to-openapi#200]
In the example code below, I define a Zod schema (`testObject`), and use the openapi extension to name it `TestObject`.
This schema is used as the response of the GET route. The PUT route body is defined to use the schema `testObject.deepPartial()`. This schema is not named separately.

In the generated OAS, the put route body refers to the same OAS schema as the get route response. This is not a correct spec; The deepPartial indicates that no property in the put route body should be required, but still they are in the OAS.

If i name the partial Zod schema separately (`testObject.deepPartial().openapi('TestObjectPartial')`), things do work as expected.
However, I'd expect the library to _not_ use the same OAS ref when the zod schema is changed after the schema is named.

In this example, just naming the schema separately is an easy fix. In a large complex codebase however, Zod are passed are passed around everywhere, in which case this might be pretty easy to miss, resulting in a bad OAS.

```ts
import { z } from 'zod';
import {
    OpenAPIRegistry,
    OpenApiGeneratorV3,
    extendZodWithOpenApi
} from '@asteasolutions/zod-to-openapi';
import { writeFileSync } from 'node:fs'
import { resolve } from 'node:path'

extendZodWithOpenApi(z)

const registry = new OpenAPIRegistry();

const testObject = z.object({
    someString: z.string(),
    someOptionalString: z.string().optional(),
    aNumber: z.number(),
}).openapi('TestObject')

registry.registerPath({
    method: 'get',
    path: '/users/{id}',
    summary: 'Get a single user',
    request: {
        params: z.object({ id: z.string() }),
    },

    responses: {
        200: {
            description: 'API Response',
            content: {
                'application/json': {
                    schema: testObject,
                },
            },
        },
    },
})

registry.registerPath({
    method: 'put',
    path: '/users/{id}',
    summary: 'Get a single user',
    request: {
        params: z.object({ id: z.string() }),
        body: {
            content: {
                'application/json': {
                    schema: testObject.deepPartial()
                }
            }
        }
    },

    responses: {
        200: {
            description: 'API Response',
            content: {
                'application/json': {
                    schema: testObject,
                },
            },
        },
    },
})

const generator = new OpenApiGeneratorV3(registry.definitions);

const openapiSpec = generator.generateDocument({
    openapi: '3.0.0',
    info: {
        version: '1.0.0',
        title: 'My API',
        description: 'This is the API',
    },
    servers: [{ url: 'v1' }],
})

writeFileSync(
    resolve(__dirname, '../openapi.json'),
    JSON.stringify(openapiSpec)
)
```

Output OpenAPI spec:

```json
{"openapi":"3.0.0","info":{"version":"1.0.0","title":"My API","description":"This is the API"},"servers":[{"url":"v1"}],"components":{"schemas":{"TestObject":{"type":"object","properties":{"someString":{"type":"string"},"someOptionalString":{"type":"string"},"aNumber":{"type":"number"}},"required":["someString","aNumber"]}},"parameters":{}},"paths":{"/users/{id}":{"get":{"summary":"Get a single user","parameters":[{"schema":{"type":"string"},"required":true,"name":"id","in":"path"}],"responses":{"200":{"description":"API Response","content":{"application/json":{"schema":{"$ref":"#/components/schemas/TestObject"}}}}}},"put":{"summary":"Get a single user","parameters":[{"schema":{"type":"string"},"required":true,"name":"id","in":"path"}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/TestObject"}}}},"responses":{"200":{"description":"API Response","content":{"application/json":{"schema":{"$ref":"#/components/schemas/TestObject"}}}}}}}}}
```

**Comment by AGalabov** [comment]
@evertdespiegeleer thank you for using our library and bringing this up. 

Yes naming it separately would for sure help in this case. And I do agree that ideally this should be treated as a new schema - will have to look into it further to figure out what causes it.

**Comment by AGalabov** [comment]
@evertdespiegeleer it was a small mistake on out own that was specific to the usage of `deepPartial`. I fixed it and added some tests around it in the linked PR.

The code is now merged and available as part of the [v6.3.1 release](https://github.com/asteasolutions/zod-to-openapi/releases/tag/v6.3.1)

**Comment by evertdespiegeleer** [comment]
Awesome! Thanks a lot for the quick fix!
